src/osm2mimir.rs

```diff
@@ -0,0 +1,183 @@
+};
+use mimir::{
+    adapters::secondary::elasticsearch::{self, ElasticsearchStorage},
+    domain::ports::{primary::generate_index::GenerateIndex, secondary::remote::Remote},
+};
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Settings (Configuration or CLI) Error: {}", source))]
+    Settings { source: settings::Error },
+
+    #[snafu(display("OSM PBF Reader Error: {}", source))]
+    OsmPbfReader { source: crate::osm_reader::Error },
+
+    #[snafu(display("Elasticsearch Connection Pool {}", source))]
+    ElasticsearchConnection {
+        source: mimir::domain::ports::secondary::remote::Error,
+    },
+
+    #[snafu(display("Street Extraction from OSM PBF Error {}", source))]
+    StreetOsmExtraction {
+        source: crate::osm_reader::street::Error,
+    },
+
+    #[snafu(display("Street Index Creation Error {}", source))]
+    StreetIndexCreation {
+        source: mimir::domain::model::error::Error,
+    },
+
+    #[snafu(display("Poi Extraction from OSM PBF Error {}", source))]
+    PoiOsmExtraction {
+        source: crate::osm_reader::poi::Error,
+    },
+
+    #[snafu(display("Poi Index Creation Error {}", source))]
+    PoiIndexCreation {
+        source: mimir::domain::model::error::Error,
+    },
+
+    #[snafu(display("Elasticsearch Configuration {}", source))]
+    StreetElasticsearchConfiguration { source: common::config::Error },
+
+    #[snafu(display("Execution Error {}", source))]
+    Execution { source: Box<dyn std::error::Error> },
+
+    #[snafu(display("Admin Retrieval Error {}", details))]
+    AdminRetrieval { details: String },
+}
+
+#[instrument(skip_all)]
+async fn import_streets(
+    streets: Vec<places::street::Street>,
+    client: &ElasticsearchStorage,
+    config: &ContainerConfig,
+) -> Result<(), Error> {
+    let streets = streets
+        .into_iter()
+        .map(|street| street.set_weight_from_admins());
+
+    let _index = client
+        .generate_index(config, futures::stream::iter(streets))
+        .await
+        .context(StreetIndexCreationSnafu)?;
+
+    Ok(())
+}
+
+#[instrument(skip_all)]
+async fn import_pois(
+    osm_reader: &mut crate::osm_reader::OsmPbfReader,
+    admins_geofinder: &AdminGeoFinder,
+    poi_config: &crate::osm_reader::poi::PoiConfig,
+    client: &ElasticsearchStorage,
+    config: &ContainerConfig,
+    max_distance_reverse: usize,
+) -> Result<(), Error> {
+    // This function rely on AdminGeoFinder::get_objs_and_deps
+    // which use all available cpu/cores to decode osm file and cannot be limited by tokio runtime
+    let pois = crate::osm_reader::poi::pois(osm_reader, poi_config, admins_geofinder)
+        .context(PoiOsmExtractionSnafu)?;
+
+    let pois: Vec<places::poi::Poi> = futures::stream::iter(pois)
+        .map(crate::osm_reader::poi::compute_weight)
+        .then(|poi| crate::osm_reader::poi::add_address(client, poi, max_distance_reverse))
+        .collect()
+        .await;
+
+    let _ = client
```

Comment:
not sure if it is a problem, but suggested by [sonarcloud](https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=819&id=Hove_mimirsbrunn&open=AYcykfatSCxZF6HnZwBe)
```suggestion
    let _client = client
```

cf https://doc.rust-lang.org/stable/nightly-rustc/rustc_lint/let_underscore/static.LET_UNDERSCORE_DROP.html

Reply:
Done in [f38e074](https://github.com/hove-io/mimirsbrunn/pull/819/commits/f38e07421bb31148a53531e584329fb89fbba381).